### PR TITLE
feat: support multiple stmts in one run (simple query)

### PR DESCRIPTION
**Pull request by BowenXiao1999**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

Test locally.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

## Refer to a related PR or issue link (optional)
Relates #6789

### CODE REVIEW

File: src/utils/pgwire/src/pg_protocol.rs [hunk 1]
```diff
@@ -312,47 +313,58 @@ where
...
-        let mut res = session
-            .run_statement(sql, false)
-            .await
-            .map_err(|err| PsqlError::QueryError(err))?;
 
-        if let Some(notice) = res.get_notice() {
-            self.stream
-                .write_no_flush(&BeMessage::NoticeResponse(&notice))?;
-        }
+        // Parse sql.
```
Comment (BowenXiao1999) [inline src/utils/pgwire/src/pg_protocol.rs:317]: Add parse sql here.

File: src/utils/pgwire/src/pg_protocol.rs [hunk 2]
```diff
@@ -311,48 +311,58 @@ where
...
+        // Execute multiple statements in simple query. KISS later.
+        for stmt in stmts {
+            if stmt.is_empty() {
+                continue;
+            }
+            let session = session.clone();
+            let mut res = session
+                .run_statement(stmt, false)
+                .await
+                .map_err(|err| PsqlError::QueryError(err))?;
```
[the remark this answers is not in the document]
Comment (BowenXiao1999) [reply src/utils/pgwire/src/pg_protocol.rs]: Well I switch backed again 😥. Cuz it's very differecult to split statement without parser. For example, `SELECT 'a;a;a;';` will be hard to split

[the remark this answers is not in the document]
Comment (xxchan) [reply src/utils/pgwire/src/pg_protocol.rs]: FYI here's how `psql` handle it: https://github.com/postgres/postgres/blob/9d0cf574920f1d5e6c260815d242b6691d37d5dc/src/fe_utils/psqlscan.l#L13-L20

🤪 I have no objection to depending on parser here. KISS...

**Review by ZENOTME — APPROVED** [review]

File: src/frontend/src/session.rs [hunk 3]
```diff
@@ -683,6 +683,36 @@ impl Session<PgResponseStream> for SessionImpl {
         Ok(rsp)
     }
 
+    /// A copy of run_statement but exclude the parser part so each run must be at most one
+    /// statement. The str sql use the to_string of AST. Consider Reuse later.
+    async fn run_one_query(
```
[the remark this answers is not in the document]
Comment (BowenXiao1999) [reply src/frontend/src/session.rs:688]: Yes. After the remove it from extended mode.

[the remark this answers is not in the document]
Comment (BowenXiao1999) [reply src/frontend/src/session.rs:688]: +1

**Comment by BowenXiao1999** [comment]
https://github.com/risingwavelabs/risingwave/issues/6801#issuecomment-1357217483

**Comment by fuyufjh** [comment]
@BowenXiao1999 Can we continue? It seems a necessary feature for some 3rd-party libs.

**Comment by BowenXiao1999** [comment]
> @BowenXiao1999 Can we continue? It seems a necessary feature for some 3rd-party libs.

Sure, previously I'm just a little afraid some other potential bugs that may brought by this pr.

File: src/utils/pgwire/src/pg_protocol.rs [hunk 4]
```diff
@@ -317,52 +318,67 @@ where
...
-        }
+        // Parse sql.
+        let stmts = Parser::parse_sql(sql)
+            .inspect_err(|e| tracing::error!("failed to parse sql:\n{}:\n{}", sql, e))
+            .map_err(|err| PsqlError::QueryError(err.into()))?;
 
-        if res.is_query() {
-            self.stream
-                .write_no_flush(&BeMessage::RowDescription(&res.get_row_desc()))?;
+        println!("simple here");
```
Comment (BugenZhao) [inline src/utils/pgwire/src/pg_protocol.rs:327]: Should we remove this line? 😰

Comment (BowenXiao1999) [reply src/utils/pgwire/src/pg_protocol.rs:327]: Sorry my bad.. Will submit a new pr...

**Pull request merged**